### PR TITLE
feat(vector tiles): Add materilized view

### PR DESCRIPTION
**Pull request by ff6347**
This PR adds a materilized view that should be used via the API to generate vector tiles.

### CODE REVIEW

File: infrastructure/database/templates/07_create_or_update_views.template.sql [hunk 1]
```diff
@@ -22,25 +22,105 @@ FROM public.radolan;
...
-		shading.winter, shading.spring, shading.summer, shading.fall,
-		trees.gattung_deutsch, trees.standalter,
-		weather_14d_agg.temp_avg_c_14d_avg, weather_14d_agg.wind_avg_ms_14d_avg, weather_14d_agg.temp_max_c_14d_avg, weather_14d_agg.wind_max_ms_14d_avg 
-FROM private.sensor_measurements
-LEFT JOIN public.shading ON public.shading.tree_id = sensor_measurements.tree_id
-LEFT JOIN public.trees ON trees.id = sensor_measurements.tree_id
-LEFT JOIN public.weather_14d_agg ON sensor_measurements.timestamp = weather_14d_agg.timestamp
-LEFT JOIN public.radolan_14d_agg ON sensor_measurements.timestamp = radolan_14d_agg.timestamp
-ORDER BY timestamp DESC;
+CREATE MATERIALIZED VIEW public.vector_tiles AS
```
Comment (birdMarcus) [inline infrastructure/database/templates/07_create_or_update_views.template.sql:33]: Is public alright?

File: infrastructure/database/templates/07_create_or_update_views.template.sql [hunk 2]
```diff
@@ -22,25 +22,105 @@ FROM public.radolan;
...
+	trees.bezirk AS trees_bezirk,
+	trees.eigentuemer AS trees_eigentuemer,
+	trees.zusatz AS trees_zusatz,
+	trees.kronedurch AS trees_kronedurch,
+	trees.geometry AS trees_geometry,
+	trees.lat AS trees_lat,
+	trees.lng AS trees_lng,
+	trees.created_at AS trees_created_at,
+	trees.updated_at AS trees_updated_at,
+	trees.street_tree AS trees_street_tree,
```
Comment (birdMarcus) [inline infrastructure/database/templates/07_create_or_update_views.template.sql:58]: should it also include Baumscheibe from latest merge in main?

**Review by birdMarcus — CHANGES_REQUESTED** [review]
Needed to get rid of quotes for Python. 
Check the two comments before merging.

File: infrastructure/database/templates/07_create_or_update_views.template.sql (hunk 1 above)
Comment (ff6347) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:33]: Yes. It should be in the public schema

File: infrastructure/database/templates/07_create_or_update_views.template.sql (hunk 2 above)
Comment (ff6347) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:58]: Sure. I'll push a fix

Comment (ff6347) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:58]: Cant see baumscheibe in the current staging env.

Comment (birdMarcus) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:58]: It's in main with this PR: https://github.com/technologiestiftung/qtrees-ai-data/pull/23

Comment (ff6347) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:58]: Ah. Okay. Are you doing PRs directly into main? This one goes into staging

Comment (birdMarcus) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:58]: Oh. Okay, maybe I missed that. Didn't know about staging.

Comment (birdMarcus) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:58]: Nico hat zuletzt aber auch nach main gemerged. Hatten wir da mal ne andere Konvention besprochen?

Comment (ff6347) [reply infrastructure/database/templates/07_create_or_update_views.template.sql:58]: I don't know. I just saw the staging branch and thought: "hey lets use staging"

**Comment by birdMarcus** [comment]
@ff6347 Ist der Training view verloren gegangen?

**Comment by ff6347** [comment]
> @ff6347 Ist der Training view verloren gegangen?

Never saw a training view. Did not remove anything. Might be that my base was wrong?
image

I will merge main into it and see what happens

**Comment by birdMarcus** [comment]
image

Der ist dann auch nicht mehr in der finalen File.

**Comment by ff6347** [comment]
Oh. Sorry. I'll add it back in

**Comment by birdMarcus** [comment]
LGTM

**Pull request merged**